app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list) -> list[Person]:
+    person_list = [Person(person_dict["name"],
+                          person_dict["age"])
+                   for person_dict in people]
+
+    for person_dict in people:
+        person = Person.people[person_dict["name"]]
+        if person_dict and person_dict.get("wife"):
```

Comment:
You just can check `if person_dict.get("wife")` here , that already will mean that dict is not empty : )